File: Roost.Web.Angular/src/app/models/Item.ts
import { Attributes } from "./attributes";
import { UnitOfMeasure } from "./unitOfMeasure";

export interface Item {
  id: string;
  itemName: string;
  itemDescription: string;
  category: string;
  categoryDescription: string;
  parentCategory: string;
  parentCategoryDescription: string;
  unitQuantity: number;
  upc: string;
  itemAttributes: Attributes;
  unitOfMeasure: UnitOfMeasure;
  imageUrl: string;
}

File: Roost.Web.Server/Models/Item.cs
using Newtonsoft.Json;

namespace Roost.Web.Server.Models
{
    public class Item
    {
        public string Upc { set; get; }

        public string ItemName { get; set; }

        public string ItemDescription { get; set; }

        public double UnitQuantity { set; get; }

        public List<ItemAttribute>? Attributes { get; set; }

        public string Category { get; set; }

        public string CategoryDescription { get; set; }

        public string? ParentCategory { get; set; }

        public string? ParentCategoryDescription { get; set; }

        public UnitOfMeasure? UnitOfMeasure { get; set; }

        public List<ItemImage>? Images { get; set; }
    }
}
